pkg/foundation/database/acceptance_testing.go

```diff
@@ -184,6 +187,79 @@ func testTransactionVisibility(t *testing.T, db DB) {
 	})
 }
 
+func testConcurrency(t *testing.T, db DB) {
```

Comment:
I added a concurrency test that ensures the implementation can handle concurrent access. We need to fix this test, then sqlite can become an option.